src/main/java/mateusz/michal/chat/Model/JsonResponseFactory.java

```diff
@@ -5,9 +5,9 @@
 import java.util.List;
 
 @Component
-public class JsonFactory {
+public class JsonResponseFactory {
 
-    public IJsonResponse createResponse(ResponseEnum responseEnum,
+    public static IJsonResponse createResponse(ResponseEnum responseEnum,
                                         List<MyError> errors,
                                         IData data,
                                         IMetaData metaData){
```

Comment:
A co powiesz na [method overloading](https://www.geeksforgeeks.org/overloading-in-java/)? Mógłbyś zdefiniować trzy przypadki: 

- `responseEnum = DATA`, przyjmująca tylko `responseEnum` oraz `data`, 
- `responseEnum = ERROR`, przyjmująca tylko `responseEnum` oraz `error`,
- ... i inne określone typy dla konkretnych responsów

Co nam to da?
Unikniemy przekazywania do metody tylu `null`i - zbędnych argumentów

Reply:
Myślałem nad tym. 
To rozwiązanie to jeden z pomysłów. 
Miałem jeszcze pomysł aby stworzyć IV metody: 
-createSimpleDataResponse(IData data)
-createSimpleErrorResponse(List<MyError> errors)
-createDataResponse(IData data, IMetaData metadada)
-createErrorResponse(List<MyError> errors, IMetaData metadata)
i w zależności od potrzeb wywoływać którąś z tych metod 
Ostatecznie wybrał opcję z jedną metodą.